crypto/src/hash.rs

```diff
@@ -237,7 +237,9 @@ impl HashType {
             let mut hash = Vec::with_capacity(self.base58check_prefix().len() + data.len());
             hash.extend(self.base58check_prefix());
             hash.extend(data);
-            Ok(hash.to_base58check())
+            Ok(hash
+                .to_base58check()
+                .unwrap_or_else(|_| unreachable!("Hash size should not exceed allowed 128 bytes")))
```

Comment:
@akoptelov 
can we just map_err to FromBytesError here?

@akoptelov 
also please, check all usage of `unreachable!()` and add message there please, why is it unreachable